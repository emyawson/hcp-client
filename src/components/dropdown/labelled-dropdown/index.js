export * from './labelled-dropdown.component';
