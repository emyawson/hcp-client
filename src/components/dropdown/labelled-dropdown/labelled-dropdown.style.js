import styled from 'styled-components';

import { colors, fontSize, spacing } from 'src/core';
import { weight } from 'src/components/fonts/weights';

export const LabelledDropdownContainerDiv = styled.div`
  display: flex;
  flex-direction: ${props => (props.labelIsInline ? 'row' : 'column')};
  align-items: ${props => (props.labelIsInline ? 'center' : 'stretch')};
`;

export const DropdownLabel = styled.label`
  color: ${colors.charcoal};
  display: flex;
  font-size: ${fontSize.p};
  font-weight: ${weight.semiBold};
  justify-content: space-between;
  padding-${props => (props.labelIsInline ? 'right' : 'bottom')}: ${
  spacing.three
};
`;
