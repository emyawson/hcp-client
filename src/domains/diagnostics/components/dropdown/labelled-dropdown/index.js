export { LabelledDropdown } from './labelled-dropdown.component';
