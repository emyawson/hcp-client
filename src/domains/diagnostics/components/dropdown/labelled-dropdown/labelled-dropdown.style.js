import styled from 'styled-components';

import { colors, fontSize, spacing } from 'src/domains/diagnostics/styles';
import { weight } from 'src/domains/diagnostics/components/fonts/weights';

export const LabelledDropdownContainerDiv = styled.div`
  display: flex;
  flex-direction: ${props => (props.labelIsInline ? 'row' : 'column')};
  align-items: ${props => (props.labelIsInline ? 'center' : 'stretch')};
`;

export const DropdownLabel = styled.label`
  color: ${colors.charcoal};
  font-size: ${fontSize.p};
  font-weight: ${weight.semiBold};
  padding-${props => (props.labelIsInline ? 'right' : 'bottom')}: ${
  spacing.three
};
`;
