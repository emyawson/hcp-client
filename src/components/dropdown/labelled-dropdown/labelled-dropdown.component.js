import React from 'react';

import { Dropdown } from 'src/components/dropdown';

import {
  DropdownLabel,
  LabelledDropdownContainerDiv,
} from './labelled-dropdown.style';

export const LabelledDropdown = ({
  disabled,
  label,
  labelIsInline,
  modelPath,
  options,
  placeholder,
  labelDecorator,
  onChange,
  ...props
}) => (
  <LabelledDropdownContainerDiv labelIsInline={labelIsInline}>
    <DropdownLabel htmlFor={modelPath} labelIsInline={labelIsInline}>
      {label}
      {labelDecorator && labelDecorator}
    </DropdownLabel>
    <Dropdown
      disabled={disabled}
      label={label}
      modelPath={modelPath}
      required
      searchable={false}
      options={options}
      onChange={onChange}
      placeholder={placeholder}
      {...props}
    />
  </LabelledDropdownContainerDiv>
);

LabelledDropdown.defaultProps = {
  disabled: false,
};
