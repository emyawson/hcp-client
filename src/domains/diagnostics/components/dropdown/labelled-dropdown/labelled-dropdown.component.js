import React from 'react';

import { Dropdown } from 'src/domains/diagnostics/components/dropdown';

import {
  DropdownLabel,
  LabelledDropdownContainerDiv,
} from './labelled-dropdown.style';

export const LabelledDropdown = ({
  disabled,
  label,
  labelIsInline,
  modelPath,
  options,
  placeholder,
  onChange,
}) => (
  <LabelledDropdownContainerDiv labelIsInline={labelIsInline}>
    <DropdownLabel htmlFor={modelPath} labelIsInline={labelIsInline}>
      {label}
    </DropdownLabel>
    <Dropdown
      disabled={disabled}
      label={label}
      modelPath={modelPath}
      required
      searchable={false}
      options={options}
      onChange={onChange}
      placeholder={placeholder}
    />
  </LabelledDropdownContainerDiv>
);

LabelledDropdown.defaultProps = {
  disabled: false,
};
