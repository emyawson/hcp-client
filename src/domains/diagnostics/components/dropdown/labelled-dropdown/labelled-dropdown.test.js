import { shallow } from 'enzyme';
import React from 'react';

import { LabelledDropdown } from './labelled-dropdown.component';

describe('labelled dropdown test suite', () => {
  test('it renders correctly', () => {
    const wrapper = shallow(
      <LabelledDropdown
        label="Drop down"
        labelIsInline
        modelPath=".dropdown"
        options={[
          { label: 'Uno', value: 1 },
          { label: 'Dos', value: 2 },
          { label: 'Tres', value: 3 },
        ]}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
