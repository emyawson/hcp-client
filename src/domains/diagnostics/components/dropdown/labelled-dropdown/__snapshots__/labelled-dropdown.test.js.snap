// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`labelled dropdown test suite it renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    disabled={false}
    label="Drop down"
    labelIsInline={true}
    modelPath=".dropdown"
    options={
      Array [
        Object {
          "label": "Uno",
          "value": 1,
        },
        Object {
          "label": "Dos",
          "value": 2,
        },
        Object {
          "label": "Tres",
          "value": 3,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.label
          htmlFor=".dropdown"
          labelIsInline={true}
        >
          Drop down
        </styled.label>,
        <Dropdown
          autosize={false}
          backspaceRemoves={false}
          clearable={false}
          disabled={false}
          id=""
          label="Drop down"
          modelPath=".dropdown"
          multi={false}
          onChange={[Function]}
          optionFontSize="0.875rem"
          options={
            Array [
              Object {
                "label": "Uno",
                "value": 1,
              },
              Object {
                "label": "Dos",
                "value": 2,
              },
              Object {
                "label": "Tres",
                "value": 3,
              },
            ]
          }
          placeholder=""
          required={true}
          searchable={false}
          selectHeight="3rem"
          selectedOptionPadding="1rem"
          simpleValue={true}
        />,
      ],
      "labelIsInline": true,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": "Drop down",
          "htmlFor": ".dropdown",
          "labelIsInline": true,
        },
        "ref": null,
        "rendered": "Drop down",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "autosize": false,
          "backspaceRemoves": false,
          "clearable": false,
          "disabled": false,
          "id": "",
          "label": "Drop down",
          "modelPath": ".dropdown",
          "multi": false,
          "onChange": [Function],
          "optionFontSize": "0.875rem",
          "options": Array [
            Object {
              "label": "Uno",
              "value": 1,
            },
            Object {
              "label": "Dos",
              "value": 2,
            },
            Object {
              "label": "Tres",
              "value": 3,
            },
          ],
          "placeholder": "",
          "required": true,
          "searchable": false,
          "selectHeight": "3rem",
          "selectedOptionPadding": "1rem",
          "simpleValue": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.label
            htmlFor=".dropdown"
            labelIsInline={true}
          >
            Drop down
          </styled.label>,
          <Dropdown
            autosize={false}
            backspaceRemoves={false}
            clearable={false}
            disabled={false}
            id=""
            label="Drop down"
            modelPath=".dropdown"
            multi={false}
            onChange={[Function]}
            optionFontSize="0.875rem"
            options={
              Array [
                Object {
                  "label": "Uno",
                  "value": 1,
                },
                Object {
                  "label": "Dos",
                  "value": 2,
                },
                Object {
                  "label": "Tres",
                  "value": 3,
                },
              ]
            }
            placeholder=""
            required={true}
            searchable={false}
            selectHeight="3rem"
            selectedOptionPadding="1rem"
            simpleValue={true}
          />,
        ],
        "labelIsInline": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "Drop down",
            "htmlFor": ".dropdown",
            "labelIsInline": true,
          },
          "ref": null,
          "rendered": "Drop down",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "autosize": false,
            "backspaceRemoves": false,
            "clearable": false,
            "disabled": false,
            "id": "",
            "label": "Drop down",
            "modelPath": ".dropdown",
            "multi": false,
            "onChange": [Function],
            "optionFontSize": "0.875rem",
            "options": Array [
              Object {
                "label": "Uno",
                "value": 1,
              },
              Object {
                "label": "Dos",
                "value": 2,
              },
              Object {
                "label": "Tres",
                "value": 3,
              },
            ],
            "placeholder": "",
            "required": true,
            "searchable": false,
            "selectHeight": "3rem",
            "selectedOptionPadding": "1rem",
            "simpleValue": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
